ee/tpmrunner/tpmrunner.go

```diff
@@ -0,0 +1,241 @@
+package tpmrunner
```

Comment:
Would you add a quick diagram/writeup in the `docs/architecture` folder for this? Since this work traverses both root and user launcher, and the implementation differs between darwin/other OSes (plus the keys are pretty important) I think it's worth documenting how the different pieces fit together. Doesn't have to be in this PR!

Reply:
I'm a little curious about this as well. I understand why we need to move the secure enclave signing. But I'm much less sure what's happening with the TPM

Reply:
@RebeccaMahany , I will write up some docs for this PR

@directionless there are 2 reasons I went ahead and made the tpm a "runner" style like secure enclave.

1. We do see errors pop up with TPM operations that I can't reproduce. I'm hoping that these have something to do with the TPM not being ready or unavailable for some reason, taking this route we'll keep trying in the background hoping it becomes available.

2. It felt cleaner to have both hardware signature mechanisms operate in the same way. Understanding that no kind of hardware signing can be expected to be available immediately after start up.